snippets/Collapse.md

```diff
@@ -0,0 +1,69 @@
+### Collapse
+
+Renders a component with collapsible content.
+
+Use the value of the `collapsed` prop to determine the initial state of the content (collapsed/expanded).
+Set the `state` of the component to the value of the `collapsed` prop (cast to a boolean value) and bind the `toggleCollapse` method to the component's context.
+Use an object, `style`, to hold the styles for individual components and their states.
+Create a method, `toggleCollapse`, which uses `Component.prototype.setState` to change the component's `state` from collapsed to expanded and vice versa.
+In the `render()` method, use a `React.Fragment` to wrap both the `<button>` that alters the component's `state` and the content of the component, passed down via `props.children`.
+Determine the appearance of the content, based on `state.collapsed` and apply the appropriate CSS rules from the `style` object.
+Finally, update the value of the `aria-expanded` attribute based on `state.collapsed` to make the component accessible.
+
+```jsx
+class Collapse extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collapsed: !!props.collapsed
+    };
+    this.style = {
+      collapsed: {
+        display: 'none'
+      },
+      expanded: {
+        display: 'block'
+      },
+      buttonStyle: {
+        display: 'block',
+        width: '100%'
+      }
+    };
+    this.toggleCollapse = this.toggleCollapse.bind(this);
+  }
+  
+  toggleCollapse() {
+    this.setState({ collapsed: !this.state.collapsed });
+  }
+  
+  render() {
+    return (
+      <React.Fragment>
```

Comment:
Shouldn't this use a div wrapping the button and context that is collapsible, not a fragment?

While a fragment should work, it'll mean that where Collapse is called will nest it's children on the parent directly, not under a div(IIRC)

Reply:
Yeah, I usually opt for not adding extra elements for no reason in components, but this might be a good case against fragments actually.

Reply:
Removing the Fragment will also probably drop expertise to `1` instead of `2` and make reading this a bit easier.